Public/Src/Engine/Scheduler/PipExecutor.cs

```diff
@@ -3659,6 +3660,18 @@ internal static void UpdateFileData(Dictionary<AbsolutePath, FileOutputData> map
                                 continue;
                             }
 
+                            if (!OperatingSystemHelper.IsUnixOS)

```

Comment:
>if [](start = 28, length = 2)

How is this case handled for normal output directories? Seems like it should be the same between the two.

Reply:
For situations other than SharedOpaque, the content in the output directory is scan and listed as files. So I think we won't encounter the symlink directory problem.

Reply:
Seems like we treat them as files for normal opaque directories (see comment here (line 614): [http://index/?query=EnumerateDynamicOutputDirectory&rightProject=BuildXL.Scheduler&file=Artifacts%5CFileContentManager.cs&line=614](http://index/?query=EnumerateDynamicOutputDirectory&rightProject=BuildXL.Scheduler&file=Artifacts%5CFileContentManager.cs&line=614)). To be consistent, seems like we do that same here.

---
In reply to: [284017124](https://github.com/microsoft/BuildXL/pull/295#discussion_r284017124) [](ancestors = 284017124)

Reply:
Thanks. The latest change achieve the consistence by disable output cache for the symlink dir and pip cache for all output directories.